roxie/ccd/ccdfile.cpp

```diff
@@ -1279,24 +1280,28 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         HashIterator h(files);
         ForEach(h)
         {
-            ILazyFileIO *f = files.mapToValue(&h.query());
-            if (f->isAlive() && f->isOpen() && f->isRemote()==remote && !f->isCopying())
+            ILazyFileIO * match = files.mapToValue(&h.query());
+            if (match->isAliveAndLink())
             {
-                unsigned age = msTick() - f->getLastAccessed();
-                if (age > maxFileAge[remote])
+                Owned<ILazyFileIO> f = match;
+                if (f->isOpen() && f->isRemote()==remote && !f->isCopying())
                 {
-                    if (traceLevel > 5)
+                    unsigned age = msTick() - f->getLastAccessed();
+                    if (age > maxFileAge[remote])
                     {
-                        // NOTE - querySource will cause the file to be opened if not already open
-                        // That's OK here, since we know the file is open and remote.
-                        // But don't be tempted to move this line outside these if's (eg. to trace the idle case)
-                        const char *fname = remote ? f->querySource()->queryFilename() : f->queryFilename();
-                        DBGLOG("Closing inactive %s file %s (last accessed %u ms ago)", remote ? "remote" : "local",  fname, age);
+                        if (traceLevel > 5)
+                        {
+                            // NOTE - querySource will cause the file to be opened if not already open
+                            // That's OK here, since we know the file is open and remote.
+                            // But don't be tempted to move this line outside these if's (eg. to trace the idle case)
+                            const char *fname = remote ? f->querySource()->queryFilename() : f->queryFilename();
+                            DBGLOG("Closing inactive %s file %s (last accessed %u ms ago)", remote ? "remote" : "local",  fname, age);
+                        }
+                        f->close();
                     }
-                    f->close();
+                    else
+                        goers.append(*f);   //Shouldn't this be linked?
```

Comment:
It is linked now once.  Do you mean add a LINK(*f) ?

Reply:
fixed